torch/lib/TH/generic/THTensorMath.c

```diff
@@ -707,84 +807,178 @@ void THTensor_(rshift)(THTensor *r_, THTensor *t, real value)
   return THError("rshift is not supported for torch.HalfTensor");
 #else
   THTensor_(resizeAs)(r_, t);
-  if (THTensor_(isContiguous)(r_) &&
-      THTensor_(isContiguous)(t) &&
-      THTensor_(nElement)(r_) == THTensor_(nElement)(t)) {
+  int64_t r_Size = THTensor_(nElement)(r_);
+  int64_t tSize = THTensor_(nElement)(t);
+  int r_Contig = THTensor_(isContiguous)(r_);
+  int tContig = THTensor_(isContiguous)(t);
+  
+  int serial_path = 0;
+  if (tSize == r_Size){
+    if (r_Contig && tContig) {
       real *tp = THTensor_(data)(t);
       real *rp = THTensor_(data)(r_);
-      int64_t sz = THTensor_(nElement)(t);
       int64_t i;
-      #pragma omp parallel for if(sz > TH_OMP_OVERHEAD_THRESHOLD * 100) private(i)
-      for (i=0; i<sz; i++) {
+      #pragma omp parallel for if(r_Size > TH_OMP_OVERHEAD_THRESHOLD * 100) private(i)
+      for (i=0; i<r_Size; i++) {
 #if defined(TH_REAL_IS_BYTE)
-          rp[i] = ((real) tp[i]) >> value;
+        rp[i] = ((real) tp[i]) >> value;
 #else
-          rp[i] = ((ureal) tp[i]) >> value;
+        rp[i] = ((ureal) tp[i]) >> value;
 #endif
       }
-  } else {
+    } else {
+#ifdef _OPENMP
+      int tZeroStride = THTensor_(hasZeroStride)(t);
+      int r_ZeroStride = THTensor_(hasZeroStride)(r_);
+      int inOMP = omp_in_parallel();
+      if (inOMP || tZeroStride || r_ZeroStride) {
+        serial_path = 1;
+      } else {
+#if defined(TH_REAL_IS_BYTE)
+        TH_TENSOR_APPLY2_ADVANCED_INDEX(r_Size, r_Contig, tContig, real, r_, real, t, *r__data = (((real) *t_data) >> value););
+#else
+        TH_TENSOR_APPLY2_ADVANCED_INDEX(r_Size, r_Contig, tContig, real, r_, real, t, *r__data = (((unsigned real) *t_data) >> value););
```

Comment:
Due to the Windows refactoring, you should now use `ureal` instead of `unsigned real`. Same in the other places in the code